packages/komencikit/src/kit.ts

```diff
@@ -0,0 +1,378 @@
+    return this.submitMetaTransaction(walletAddress, attestations.selectIssuers(identifier))
+  }
+
+  /**
+   * completeAttestation: just wraps the `submitMetaTransaction` action in order
+   * to execute Attestations.complete(identifier, walletAddress, issuer, code)
+   * TODO: I'm not sure if we need this. Valora can call submitMetaTransaction
+   *
+   * @param identifier - the phone number identifier
+   * @param walletAddress - MetaTransactionWallet address requesting attestations
+   * @param issuer - the issuer ID
+   * @param code - the code
+   */
+  public async completeAttestation(
+    identifier: string,
+    walletAddress: string,
+    issuer: Address,
+    code: string
+  ): Promise<Result<TransactionReceipt, FetchError | TxError>> {
+    const attestations = await this.contractKit.contracts.getAttestations()
+    return this.submitMetaTransaction(
+      walletAddress,
+      await attestations.complete(identifier, walletAddress, issuer, code)
+    )
+  }
+
+  /**
+   * submitMetaTransaction: uses the /v1/submitMetaTransaction endpoint
+   * It receives a wallet address and transaction (as a CeloTransactionObject)
+   * and creates a signature and passes everything to Komenci for execution
+   *
+   * @param walletAddress - the MTW that should execute the transaction
+   * @param tx - the transaction to be executed
+   * @param nonce - optional nonce to be used for signing the meta-tx
+   */
+  @retry({
+    tries: 3,
+    bailOnErrorTypes: [
+      FetchErrorTypes.Unauthorised,
+      FetchErrorTypes.ServiceUnavailable,
+      TxErrorTypes.Revert,
+    ],
+    onRetry: (_args, error, attempt) => {
+      console.debug(`${TAG}/submitMetaTransaction attempt#${attempt} error: `, error)
+    },
+  })
+  public async submitMetaTransaction(
+    walletAddress: string,
+    tx: CeloTransactionObject<any>,
+    nonce?: number
+  ): Promise<Result<TransactionReceipt, FetchError | TxError>> {
+    const wallet = await this.getWallet(walletAddress)
+    const signature = await wallet.signMetaTransaction(tx.txo, nonce)
+    const rawMetaTx = toRawTransaction(wallet.executeMetaTransaction(tx.txo, signature).txo)
+
+    const resp = await this.client.exec(submitMetaTransaction(rawMetaTx))
+    if (resp.ok) {
+      const txHash = resp.result.txHash
+      return this.waitForReceipt(txHash)
+    }
+
+    return resp
+  }
+
+  /**
+   * Utility function used to wait for a transaction to finalise and return the receipt
+   *
+   * @param txHash - the hash of the transaction to watch
+   * @private
+   */
+  private async waitForReceipt(txHash: string): Promise<Result<TransactionReceipt, TxError>> {
+    let receipt: TransactionReceipt | null = null
+    let waited = 0
+    while (receipt == null && waited < this.options.txRetryTimeoutMs) {
+      receipt = await this.contractKit.web3.eth.getTransactionReceipt(txHash)
+      if (receipt == null) {
+        await sleep(100)
```

Comment:
We can also make this a property under KomenciOptions: `txRetryPollingIntervalMs`